test/unit/app-lifecycle.spec.js

```diff
@@ -99,6 +99,44 @@ describe('App-Lifecycle', function() {
     });
   });
 
+  describe('when restarting the app', function() {
+    beforeEach(function() {
+      this.beforeStartSpy = sinon.spy(function(options) {
```

Comment:
spy is better when you want to watch something that exists, where stub is better when you want to create something to observe of act a specific way.

But rather than returning values using `calledWith` will give better results if the test fails.  `to.be.true` will show something like "test description expected true but got false" where as the `calledWith` will show "test description expected to be called with `{ foo: 'baz' }` but got `{ foo: 'bar' }`"